server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -118,33 +119,46 @@
   /**
    * Used for testing only
    */
-  protected static void clearLoadedModules()
+  static void clearLoadedModules()
   {
     extensionsMap.clear();
   }
 
+  /**
+   * Used for testing only
+   */
+  static Map<String, URLClassLoader> getLoadersMap()
+  {
+    return loadersMap;
+  }
+
+  /**
+   * Look for extension modules for the given class from both classpath and druid.extensions.coordinates.
+   * Extensions explicitly specified in druid.extensions.coordinates will be loaded first, if there is a duplicate
+   * extension from classpath, it will be ignored.
+   *
+   * @param config Extensions configuration
+   * @param clazz The class of extension module (e.g., DruidModule)
+   * @return
+   */
   public synchronized static <T> Collection<T> getFromExtensions(ExtensionsConfig config, Class<T> clazz)
   {
     final TeslaAether aether = getAetherClient(config);
-    Set<T> retVal = Sets.newHashSet();
-
-    if (config.searchCurrentClassloader()) {
-      for (T module : ServiceLoader.load(clazz, Initialization.class.getClassLoader())) {
-        log.info("Adding local module[%s]", module.getClass());
-        retVal.add(module);
-      }
-    }
+    final Set<T> retVal = Sets.newHashSet();
+    final Set<String> extensionNames = Sets.newHashSet();
 
     for (String coordinate : config.getCoordinates()) {
       log.info("Loading extension[%s] for class[%s]", coordinate, clazz.getName());
```

Comment:
@guobingkun ok, is there a reason not to use the canonical name instead of getName()? I feel the canonical name is more user-friendly for logging. I don't think that should break anything.